src/Button/variantStyles.js

```diff
@@ -21,6 +21,17 @@ const variants = ({ intent, variant, paletteColor, ...props }) => {
   const colorActive = shade(0.16, colorIntent);
   const colorFocus = themeGet(`colors.intent.${intent}.light`)(props);
   const colorHover = tint(0.16, colorIntent);
+  const colorIcon = () => {
+    if (paletteColor) {
+      return themeGet(`colors.palette.${paletteColor}.default`)(props);
+    }
+
+    if (variant !== 'primary' && intent === 'brand') {
+      return themeGet(`colors.icon.default`)(props);
```

Comment:
here is the fix. we use the default icon color if palette override doesn't exist.